# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

In [1]:
import numpy as np

In [2]:
f = lambda x: x**3

In [3]:
def calcular_area(a:int, b:int, func:callable, n_ret:int) -> float:
    delta = (b - a) / n_ret
    x_cols = np.linspace(a, b - delta, n_ret) + delta / 2
    y_cols = func(x_cols)

    print(f'Número de retângulos = {n_ret}')
    print(f'Delta = {delta}')

    return sum(delta * y_cols)

In [4]:
calcular_area(a=1, b=2, func=f, n_ret=4)

Número de retângulos = 4
Delta = 0.25


3.7265625

In [5]:
calcular_area(a=1, b=2, func=f, n_ret=8)

Número de retângulos = 8
Delta = 0.125


3.744140625

In [6]:
calcular_area(a=1, b=2, func=f, n_ret=100)

Número de retângulos = 100
Delta = 0.01


3.7499624999999996

2. Calcule essa área de forma analítica.



$$\int_{a}^{b} f(x) dx = F(b) - F(a)$$

$$ f(x) = x^3 \rightarrow F(x) = \frac{1}{4} * x^4 + C,$$

$$\int_{1}^{2} x^3 dx = F(2) - F(1) = \left(\frac{1}{4} * 2^4\right) - \left(\frac{1}{4} * 1^4\right) = \frac{16}{4} - \frac{1}{4} = \frac{15}{4} = \textbf{3.75}$$